### PR TITLE
[DRAFT] UI redesign

### LINKED ISSUE #7437: Top Right Navbar is wrapping incorrectly on smaller screens
image

### LINKED ISSUE #9220: Header isn't fully responsive at smaller browser widths
Everything fits at wide widths:
Image

For a narrower width, the links render a bit strangely:

Image

Image

### LINKED ISSUE #9223: The Next.js site can save the users's desired SDK in state as they navigate back and forth between SDK-specific docs and generic product docs

### LINKED ISSUE #9369: Merge the Platform + Guide selector
As a first step of getting closer to our redesign, we wanna merge the current platform dropdown with the guide sidebar (on the right side), and move the new (combined) dropdown to the left sidebar.

Screenshot 2024-03-06 at 18 34 09

The design changes are currently documented in [Figma](https://www.figma.com/file/8tfqsq81AJQbc04IvZHa6X/docs-redesign-'23-'24?node-id=109%3A5205&mode=dev). But the new platform/guide dropdown will move into the left sidebar, instead of the top navigation (as shown in Figma). The design / functionality of the dropdown will be very similar to this:

Screenshot 2024-03-06 at 18 37 25

### LINKED ISSUE #9423: Platform change doesn't remember the URL path
### Steps to Reproduce

1. Go to the JS SDK [custom instrumentation page](https://docs.sentry.io/platforms/javascript/performance/instrumentation/custom-instrumentation/).
2. Change the platform to e.g. Python.

### Expected Result

Get redirected to the Python SDK [custom instrumentation page](https://docs.sentry.io/platforms/python/performance/instrumentation/custom-instrumentation/).

### Actual Result

Get redirected to the Python SDK [getting started page](https://docs.sentry.io/platforms/python/).

### Additional Info

Switching the platform back to JS keeps me on the getting started page.

### LINKED ISSUE #9656: Improve 404 page
### Problem Statement

If you go to a page that does not exist, e.g.:
https://docs.sentry.io/platforms/javascript/DOES-NOT-EXIST

You get a very generic 404 page. It would be nice to show the 404 page in the context of the platform/guide that is in the URL - this makes it much easier to continue from a broken link.

### Solution Brainstorm

Something like this(formatting is off, but you get the point hopefully xD)

image

### LINKED ISSUE #9694: .NET platform menu wraps line due to long names
image

Could we get a bit larger max width to avoid this?

### PR DESCRIPTION
**Pull request by a-hariti**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

a wip branch to implement the new design

closes #9369
closes #9694 
closes #9656
closes #9423
closes #9223
closes #9220
closes #7437

### CODE REVIEW

**Comment by lizokm** [comment]
@a-hariti this is a good first step! 

Here are some changes we'd need to make in order of priority:
- we would like to see an option of the platform dropdown that has both the languages and the frameworks as in this image: 
Screenshot 2024-03-13 at 12 40 25 PM
- we want to get rid of the separate product nav and move it so that it lives persistently even when users choose their SDK. (As a next step we'll need to reorganize the folder structure of the product docs, but we can do that a bit later.) 
Screenshot 2024-03-13 at 12 55 58 PM
- Python frameworks currently live in the "integration" section of docs, we'd need to make sure they are also visible in the dropdown.
Screenshot 2024-03-13 at 12 58 51 PM
- Would like to see an option where there are 2 dropdown selectors, one for language and another for framework:
Screenshot 2024-03-13 at 1 04 29 PM

We don't need to worry about this right now, but of course we'd also need to update the colors/font styles etc. to reflect the new design.
-
Screenshot 2024-03-13 at 1 12 56 PM

**Comment by a-hariti** [comment]
Can the nested dropdown and two drop downs (for platforms  & guides) coexist @lizokm ?

**Comment by lizokm** [comment]
@a-hariti I don't think it makes sense for the nested dropdown and the two drop downs (for platforms & guides) to coexist. It should be one or the other. 

Also, I resolved a conflict that may have led to more conflicts in the sidenav. Are you able to see the previews?

**Comment by a-hariti** [comment]
Yes, I can see the previews.

Thank you. I think you accidentally left a foreign string during the conflict resolution 🙊, I'll take care of it

**Comment by a-hariti** [comment]
@lizokm should the python integrations be converted to guides in the docs or should we just change to the way they are represented in the UI?

The second option will add some hardcoded logic for python in the code base

**Comment by lizokm** [comment]
@stephanie-anderson @antonpirker would love your thoughts here: should the python integrations be converted to guides in the docs or should we just change to the way they are represented in the UI?

The second option will add some hardcoded logic for python in the code base

File: src/middleware.ts [hunk 1]
```diff
@@ -31,6 +31,30 @@ const handleRedirects = (request: NextRequest) => {
 type PathWithTrailingSlash = `${string}/`;
 
 const REDIRECTS: {from: PathWithTrailingSlash; to: string}[] = [
+  {
```
Comment (a-hariti) [inline src/middleware.ts:37]: What should we keep from these duplicates (6 entries, 3 actual redirects)?
@lizokm  @stephanie-anderson

File: src/middleware.ts [hunk 2]
```diff
@@ -31,6 +31,30 @@ const handleRedirects = (request: NextRequest) => {
...
+    from: '/product/data-management-settings/dynamic-sampling/',
+    to: '/product/accounts/quotas/',
+  },
+  {
+    from: '/platforms/data-management/',
+    to: '/product/data-management-settings/scrubbing/advanced-datascrubbing/',
+  },
+  {
+    from: '/platforms/data-management/',
+    to: '/product/data-management-settings/data-forwarding/',
```
Comment (stephanie-anderson) [inline src/middleware.ts:51]: to: '/concepts/data-management/',

File: src/middleware.ts [hunk 3]
```diff
@@ -31,6 +31,30 @@ const handleRedirects = (request: NextRequest) => {
...
+    to: '/product/performance/retention-priorities/',
+  },
+  {
+    from: '/product/data-management-settings/dynamic-sampling/',
+    to: '/product/accounts/quotas/',
+  },
+  {
+    from: '/platforms/data-management/',
+    to: '/product/data-management-settings/scrubbing/advanced-datascrubbing/',
+  },
```
Comment (stephanie-anderson) [inline src/middleware.ts:48]: remove

File: src/middleware.ts [hunk 4]
```diff
@@ -31,6 +31,30 @@ const handleRedirects = (request: NextRequest) => {
 type PathWithTrailingSlash = `${string}/`;
 
 const REDIRECTS: {from: PathWithTrailingSlash; to: string}[] = [
+  {
+    from: '/product/data-management-settings/dynamic-sampling/',
+    to: '/product/performance/retention-priorities/',
```
Comment (stephanie-anderson) [inline src/middleware.ts:39]: to: '/product/performance/',

File: src/middleware.ts [hunk 5]
```diff
@@ -31,6 +31,30 @@ const handleRedirects = (request: NextRequest) => {
...
+    to: '/product/data-management-settings/data-forwarding/',
+  },
+  {
+    from: '/platforms/javascript/sourcemaps/troubleshooting/',
+    to: '/platforms/javascript/sourcemaps/',
+  },
+  {
+    from: '/platforms/javascript/sourcemaps/troubleshooting/',
+    to: '/platforms/javascript/sourcemaps/troubleshooting_js/',
+  },
```
Comment (stephanie-anderson) [inline src/middleware.ts:60]: remove

File: src/middleware.ts [hunk 6]
```diff
@@ -31,6 +31,30 @@ const handleRedirects = (request: NextRequest) => {
...
 
 const REDIRECTS: {from: PathWithTrailingSlash; to: string}[] = [
+  {
+    from: '/product/data-management-settings/dynamic-sampling/',
+    to: '/product/performance/retention-priorities/',
+  },
+  {
+    from: '/product/data-management-settings/dynamic-sampling/',
+    to: '/product/accounts/quotas/',
+  },
```
Comment (stephanie-anderson) [inline src/middleware.ts:44]: remove

**Review by stephanie-anderson — APPROVED** [review]
let's do it

**Pull request merged**